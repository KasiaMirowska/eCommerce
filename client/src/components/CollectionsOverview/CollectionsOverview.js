import React from 'react';
import { connect } from 'react-redux';
import {selectCollectionsForPreview} from '../../redux/shop/shop.selectors';
import './CollectionsOverview.styles.scss';
import CollectionPreview from '../../components/Collection/CollectionPreview';

const mapStateToProps = state => ({
    collections: selectCollectionsForPreview(state),
})

export const CollectionsOverview = (props) => {
    const list = props.collections.map(collection => (
        <CollectionPreview
            key={collection.id}
            title={collection.title}
            items={collection.items}
            route={collection.routeName}
        />
    ))
    return (
        <div className='collection-overview'>
            {list}
        </div>
    )
}

export default connect(mapStateToProps )(CollectionsOverview);