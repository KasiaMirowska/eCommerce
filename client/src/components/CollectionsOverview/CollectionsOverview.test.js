import React from 'react';
import { shallow, mount } from 'enzyme';
import { CollectionsOverview } from '../../components/CollectionsOverview/CollectionsOverview';
import CollectionPreview from '../../components/Collection/CollectionPreview';


describe('CollectionsOverview', () => {
    const props = {
        collections: [{col: 1}, {col: 2}, {col: 3}, {col: 4}]
    }
    const wrapper = shallow(<CollectionsOverview {...props} />);

    it('renders as expected', () => {
        const div = document.createElement('div');
        expect(wrapper).toMatchSnapshot();
    });

    it('displays all collections', () => {
        expect(wrapper.find(CollectionPreview).length).toEqual(props.collections.length);
    })
})