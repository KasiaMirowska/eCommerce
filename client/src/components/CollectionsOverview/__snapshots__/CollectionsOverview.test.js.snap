// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CollectionsOverview renders as expected 1`] = `ShallowWrapper {}`;
